src/solc_0.8/polygon/root/land/LandTunnelV2.sol

```diff
@@ -0,0 +1,124 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.2;
+
+import "../../../common/fx-portal/FxBaseRootTunnelUpgradeable.sol";
+import "../../../common/interfaces/ILandTokenWithIsSuperOperator.sol";
+import "../../../common/interfaces/IERC721MandatoryTokenReceiver.sol";
+import "../../../common/BaseWithStorage/ERC2771Handler.sol";
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/security/PausableUpgradeable.sol";
+
+/// @title LAND bridge on L1
+contract LandTunnelV2 is
+    FxBaseRootTunnelUpgradeable,
+    IERC721MandatoryTokenReceiver,
+    ERC2771Handler,
+    OwnableUpgradeable,
+    PausableUpgradeable
+{
+    address public rootToken;
+    bool internal transferringToL2;
+
+    event Deposit(address indexed user, uint256 size, uint256 x, uint256 y, bytes data);
+    event Withdraw(address indexed user, uint256 size, uint256 x, uint256 y, bytes data);
+
+    function initialize(
+        address _checkpointManager,
+        address _fxRoot,
+        address _rootToken,
+        address _trustedForwarder
+    ) public initializer {
+        rootToken = _rootToken;
+        __Ownable_init();
+        __Pausable_init();
+        __FxBaseRootTunnelUpgradeable_initialize(_checkpointManager, _fxRoot);
+        __ERC2771Handler_initialize(_trustedForwarder);
+    }
+
+    /// @dev called on ERC721 trasnfer to this contract
+    /// @param operator address of the one sending the ERC721 Token
+    /// @return onERC721Received function selector
+    function onERC721Received(
+        address operator,
+        address, /* from */
+        uint256, /* tokenId */
+        bytes calldata /* data */
+    ) external view override returns (bytes4) {
+        require(
+            transferringToL2 || ILandTokenWithIsSuperOperator(rootToken).isSuperOperator(operator),
```

Comment:
ILandTokenWithIsSuperOperator should be the interface of `WithSuperOperators` and not the interface of land + superoperator:

ILandTokenWithIsSuperOperator > ISuperOperator

Reply:
changed to ILandTokenV2 , with both exists and superoperator function.